ttnn/cpp/ttnn/operations/conv2d/conv2d.hpp

```diff
@@ -101,7 +104,7 @@ uint32_t find_closest_common_largest_divisor(uint32_t num1, uint32_t num2, uint3
 
 template <typename T>
 ParallelConfig determine_parallel_config(
-    bool height_sharding,
+    const Conv2dConfig& conv_config,
```

Comment:
feel weird that you are passing in now whole config and you just need info about sharding, I guess because you coupled to is_***_sharding as we member function of the config struct

Reply:
Yes. That is correct. 
I implemented those functions as I wanted a shorter syntax than `shard_layout == TensorMemoryLayout::HEIGHT_SHARDED`.

Another option would be to implement those functions in TensorMemoryLayout instead of Conv2dConfig.